### PR TITLE
DRGN-635 Add api nicknames on creation/updating

### PR DESCRIPTION
**Pull request by wolfpack94**
For listing purposes, friendly names can be given to keys generated through the Dragonchain API

### CODE REVIEW

File: src/services/dragonchain-client/DragonchainClient.ts [hunk 1]
```diff
@@ -183,6 +190,23 @@ export class DragonchainClient {
...
+   */
+  public updateApiKey = async (options: {
+    /**
+     * Key ID to modify
+     */
+    keyId: string,
+    /**
+     * New nickname to set for key
+     */
+    nickname: string
```
Comment (rdaquilante) [inline src/services/dragonchain-client/DragonchainClient.ts]: Shouldn't this be optional?

File: src/interfaces/DragonchainClientInterfaces.ts [hunk 2]
```diff
@@ -263,6 +264,7 @@ export interface DeleteAPIKeyResponse {
 export interface GetAPIKeyResponse {
   id: string,
   registration_time: number,
+  nickname: string,
```
Comment (rdaquilante) [inline src/interfaces/DragonchainClientInterfaces.ts]: Will this cause issues for existing keys that don't have nicknames?

Comment (regan-karlewicz) [reply src/interfaces/DragonchainClientInterfaces.ts]: Typescript doesn't throw errors for typing at runtime

File: src/interfaces/DragonchainClientInterfaces.ts [hunk 3]
```diff
@@ -235,6 +235,7 @@ export interface CreateAPIKeyResponse {
   key: string,
   id: string,
   registration_time: number,
+  nickname: string,
```
Comment (cheeseandcereal) [inline src/interfaces/DragonchainClientInterfaces.ts]: This field won't always be there, and these types are used for autodocs. So this field should probably be marked as optional.

File: src/interfaces/DragonchainClientInterfaces.ts (hunk 2 above)
Comment (cheeseandcereal) [reply src/interfaces/DragonchainClientInterfaces.ts]: but we use it for autodocs, and only new or updated keys will get the field. This should be marked optional for sure.

File: src/services/dragonchain-client/DragonchainClient.ts (hunk 1 above)
Comment (cheeseandcereal) [reply src/services/dragonchain-client/DragonchainClient.ts]: I think the only action that updating an api key actually does is change the nickname, so this actually makes sense.

File: src/interfaces/DragonchainClientInterfaces.ts (hunk 3 above)
Comment (wolfpack94) [reply src/interfaces/DragonchainClientInterfaces.ts]: good call

**Pull request merged**